packages/key4hep-stack/package.py

```diff
@@ -1,90 +1,225 @@
 from spack import *
 from spack.package import PackageBase
+from datetime import datetime
+
 
 class Key4hepStack(BundlePackage):
-    """Bundle package to install the Key4hep software development environment."""
-    
+    """Bundle package to install the Key4hep software stack."""
     
     homepage = 'https://cern.ch/key4hep'
-    
-    version('0.1')
-    version('nightly')
-    
 
-    depends_on("cmake")
-    depends_on("gdb")
+    ##################### versions ########################
+    #######################################################
+    # tries to build the HEAD of each package.
+    # used for master builds
+    version('master')
+    # builds the latest stable version of each package
+    # the preferred usage is to use the date as version, see below
+    version(datetime.today().strftime('%Y-%m-%d'))
+    #version('2020-10-06') # example, no need to add them here
 
+    ##################### variants ########################
+    #######################################################
+    variant('devtools', default=True,
+            description="add tools necessary for software masterment to the stack")
+    variant('generators', default=True,
+            description="add some standalone generators to the stack")
+    variant('fccsw', default=True,
+            description="build fcc packages")
+    variant('ilcsoft', default=True,
+            description="build ilcsoft packages")
+    variant('bootstrap', default=True,
+            description="install some spack setup tools")
```

Comment:
I really like this idea of having different variants in general. However, I am wondering if it scales in the end, especially in the cases, where the different variants start to overlap. As long as there are no overlaps between the variants we can simply do

```python
depends_on('fcc_package', when='+fccsw')
depends_on('ilc_package', when='+ilcsoft')
```

but as far as I know there is no easy way to say that a given package should be built when either of the variants is active. It seems to be possible to do something like this:
```python
depends_on('package', when='+fccsw')
depends_on('package', when='+ilcsoft')
```

This gets even more verbose if we want to tie in the `@master` "variant", as we would have to define all the versions a second time with `@master`.

Or do we simply consider packages that would be shared by more than one variant to be always built?

Reply:
Yes, I'm actually not sure anymore these variants are needed at all -- I propose to move this issue to a new PR.

> Or do we simply consider packages that would be shared by more than one variant to be always built?

Yes, in that case I would have just dropped the `when`.